docs/src/main/asciidoc/_app-dev-guide/assemblies/modules/development/helloworld/pom.xml

```diff
@@ -0,0 +1,140 @@
+
+  <properties>
+	  <version.server>8.0.0.GA-redhat-00008</version.server>
+	  <version.bom.ee>${version.server}</version.bom.ee>
+	  <version.plugin.wildfly>4.2.0.Final</version.plugin.wildfly>
+  	  <maven.compiler.source>11</maven.compiler.source>
+          <maven.compiler.target>11</maven.compiler.target>
+  </properties>
+
+	<repositories>
+	    <repository>                                                                
+		<id>jboss-public-maven-repository</id>
+		<name>JBoss Public Maven Repository</name>
+		<url>https://repository.jboss.org/nexus/content/groups/public/</url>
+		<releases>
+		     <enabled>true</enabled>
+		     <updatePolicy>never</updatePolicy>
+		 </releases>
+		 <snapshots>
+		     <enabled>true</enabled>
+		     <updatePolicy>never</updatePolicy>
+		 </snapshots>
+		     <layout>default</layout>
+	    </repository>
+	    <repository>                                                                
+		<id>redhat-ga-maven-repository</id>
+		<name>Red Hat GA Maven Repository</name>
+		<url>https://maven.repository.redhat.com/ga/</url>
+		<releases>
+		    <enabled>true</enabled>
+		    <updatePolicy>never</updatePolicy>
+		</releases>
+		<snapshots>
+		    <enabled>true</enabled>
+		    <updatePolicy>never</updatePolicy>
+		</snapshots>
+		<layout>default</layout>
+	    </repository>
+	</repositories>
+
+	<pluginRepositories>
+	    <pluginRepository>
+		<id>jboss-public-maven-repository</id>
+		<name>JBoss Public Maven Repository</name>
+		<url>https://repository.jboss.org/nexus/content/groups/public/</url>
+		<releases>
+		    <enabled>true</enabled>
+		</releases>
+		<snapshots>
+		    <enabled>true</enabled>
+		</snapshots>
+	    </pluginRepository>
+	    <pluginRepository>
+		<id>redhat-ga-maven-repository</id>
+		<name>Red Hat GA Maven Repository</name>
+		<url>https://maven.repository.redhat.com/ga/</url>
+		<releases>
+		    <enabled>true</enabled>
+		</releases>
+		<snapshots>
+		        <enabled>true</enabled>
+		</snapshots>
+	    </pluginRepository>
+	</pluginRepositories>
+
+	<dependencyManagement>
+	    <dependencies>
+		<dependency>
+		    <groupId>org.jboss.bom</groupId>                
+		    <artifactId>jboss-eap-ee-with-tools</artifactId>
+		    <version>${version.bom.ee}</version>
+		    <type>pom</type>
+		    <scope>import</scope>
+		</dependency>
+	    </dependencies>
+	</dependencyManagement>
+	<dependencies>
+	  <dependency>                                             
+	    <groupId>jakarta.servlet</groupId>
+	    <artifactId>jakarta.servlet-api</artifactId>
+	    <scope>provided</scope>                                
+	  </dependency>
+	  <dependency>
+	    <groupId>junit</groupId>
+	    <artifactId>junit</artifactId>
+	    <scope>test</scope>
+	  </dependency>
+	</dependencies>
+	<build>
+	    <pluginManagement>
+	      <plugins>
+		<plugin>    
+		  <groupId>org.wildfly.plugins</groupId>
+		  <artifactId>wildfly-maven-plugin</artifactId>
+		  <version>${version.plugin.wildfly}</version>
```

Comment:
@emmartins Techinically `wildfly:dev` can be used, but it will provision a full server because the provisioning configuration is in a profile. If it's defined in the default configuration, then all goals that support provisioning wil provision the same server.

Reply:
Doesn't that also means we will send such config to wildfly:deploy, which is a bit wrong? Doesn't `mvn wildfly:dev -Pprovisioned-server` works as expected?

Reply:
Only goals that use the configuration will use it. The `deploy`goal doesn't provision anything so the configuration will just be ignored.

It would work with `mvn wildfly:dev -Pprovisioned-server`, but that's just a lot to type :) 

Note this is just my opinion. Thinking from a user perspective I just want to type `mvn wildfly:dev`, make a change and see my changes. Do note that changes to the POM do not take affect and require a restart. Though we should revisit that in the plugin.

Reply:
Honestly think I would prefer that "a lot to type" for now:
1. ATM we match what we use in Quickstarts (and it is a goal to have this in sync with helloworld, which serves as ready to use example of this) 
2. Provisioning config is different for profiles provisioned-server and openshift, which means we would need to reconfig. I guess in the future we  may both in sync.

Reply:
@jamezp ?

Reply:
As long as we sync it in the future. We just went through a massive refactor of the quickstarts. I'd really like to not have to do that again. Really it's up to you though @emmartins. These are just my opinions and keeping things as up to date as possible makes less work in the future and less things we'll forget to do :)

Reply:
Thanks. For the record all the followup tasks of QS are not to be done again in team work, those are small adjustments me or a volunteer will do.